src/worker/pbs_info.rs

```diff
@@ -0,0 +1,53 @@
+use std::process::Command;
+use std::{str, env};
+use regex::Regex;
+use std::time::Duration;
+
+type TimeNumberFormat = u64;
+
+fn get_value_from_qstat_string(command_output: &String, property_name: &mut String, data_pattern: &String) -> String {
+    if data_pattern.is_empty() == false {
+        property_name.push_str(data_pattern);
+        let mut re = Regex::new(property_name.as_str()).ok().unwrap();
+        let mut walltime = re.find(&command_output).unwrap();
+        re = Regex::new(data_pattern.as_str()).ok().unwrap();
+        walltime = re.find(walltime.as_str()).unwrap();
+        String::from(walltime.as_str())
+    } else {
+        return String::new();
+    }
+}
+
+fn process_time_format(raw_time: String) -> TimeNumberFormat {
+    let numbers_str: Vec<&str> = raw_time.split(":").collect();
+    let numbers: Vec<TimeNumberFormat> = numbers_str.iter()
+        .map(|x| x.parse::<TimeNumberFormat>().unwrap())
+        .collect();
+    let seconds = numbers[0] * 3600 + numbers[1] * 60 + numbers[2];
+    seconds
+}
+
+pub fn get_valltime() -> Duration {
+    let job_id = match env::var("PBS_JOBID") {
+        Result::Ok(x) => x,
+        _ => String::new(),
+    };
+    let job_id = if job_id.is_empty() == false { job_id.as_str() } else { panic!("no jobID"); };
+
+    let command_output = match Command::new("qstat")
+        .args(&["-f", job_id])
+        .output() {
+        Result::Ok(x) => match String::from_utf8(x.stdout) {
+            Ok(x) => x,
+            Err(e) => panic!("{}", e),
```

Comment:
We definitely do not want to panic here, the function should return `Result<Duration>` and handle errors properly.